cetmix_tower_yaml/wizards/cx_tower_yaml_import_wiz_upload.py

```diff
@@ -0,0 +1,120 @@
+
+        Args:
+            decoded_file (Text): YAML code
+            model_name (Char): Name of the Odoo model, if parsed from YAML
+            record_id (_type_): ID of the record, if passed from YAML
+
+        Returns:
+            Action Window: Action to open the import wizard
+        """
+
+        decoded_file, model_name, record_id = self._extract_yaml_data()
+
+        import_wizard = self.env["cx.tower.yaml.import.wiz"].create(
+            {
+                "yaml_code": decoded_file,
+                "model_name": model_name,
+                "record_id": record_id,
+            }
+        )
+
+        return {
+            "type": "ir.actions.act_window",
+            "res_model": "cx.tower.yaml.import.wiz",
+            "res_id": import_wizard.id,
+            "view_mode": "form",
+            "target": "new",
+        }
+
+    def _extract_yaml_data(self):
+        """Extract data form YAML file and validate them
+
+        Returns:
+            decoded_file, model_name, record_id
+            - decoded_file (Text): YAML code
+            - model_name (Char): Name of the Odoo model, if resolved
+            - record_id (_type_): ID of the record, if resolved
+
+        """
+
+        self.ensure_one()
+
+        # Decode base64 file
+        try:
+            decoded_file = base64.b64decode(self.yaml_file).decode("utf-8")
+        except UnicodeDecodeError as e:
+            raise ValidationError(_("Invalid YAML file")) from e
+
+        # Parse YAML file
+        try:
+            yaml_data = yaml.safe_load(decoded_file)
+        except yaml.YAMLError as e:
+            raise ValidationError(_("Invalid YAML file")) from e
+
+        if not yaml_data:
+            raise ValidationError(_("Invalid YAML file"))
+
+        # Get Odoo model name from YAML
+        yaml_model = yaml_data.get("cetmix_tower_model")
+        if not yaml_model:
+            raise ValidationError(
+                _("No model for import is specified in the YAML file")
+            )
+
+        model_name = f"cx.tower.{yaml_model.replace('_', '.')}"
+
+        # Check if model exists
+        try:
+            model = self.env[model_name]
+        except KeyError as e:
+            raise ValidationError(_("Invalid model specified in the YAML file")) from e
+
+        # Check if model supports YAML import
+        if not hasattr(model, "yaml_code"):
+            raise ValidationError(_("Model does not support YAML import"))
+
+        # Check if YAML version is supported
+        yaml_version = yaml_data.get("cetmix_tower_yaml_version")
+        if (
+            yaml_version
+            and isinstance(yaml_version, int)
+            and yaml_version > model.CETMIX_TOWER_YAML_VERSION
+        ):
+            raise ValidationError(
+                _(
+                    "YAML file version is not supported."
+                    " You may need to update the Cetmix Tower Yaml module."
+                )
+            )
```

Comment:
YAML version is a recommended, not a mandatory key. Its absence shouldn't block the import process.
However it's presence allows to avoid possible mistakes due to the data structure modifications.